### ISSUE TITLE
Provide decorators for skipping code if an optimizer is not available

### ISSUE THREAD

**Issue by nwu63** [mdolab/pyoptsparse#224]
# Description
Dymos recently added a decorator in openmdao/dymos#570, which may be something we want to implement in pyOptSparse natively also. This could be useful for:
- skipTest in pyOptSparse tests
- test scripts in other repos (e.g. MACH-Aero, pygeo)

May be related to #187.

**Comment by robfalck** [comment]
Feel free to take as much of our implementation as you want.

**Comment by marcomangano** [comment]
Has this been addressed to the extent we wanted?

**Comment by nwu63** [comment]
Not quite, but it's close. Right now we can check the availability of top-level packages with the decorator in baseclasses, but not submodules, e.g. `pySNOPT`. The intent here is to check specifically for the optimizers, which in most cases involves checking the compiled library, e.g. `snopt.so`.
